pkgs/by-name/wo/worldpainter/package.nix

```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  jre,
+  gnused,
+}:
+stdenv.mkDerivation rec {
+  pname = "worldpainter";
+  version = "2.23.2";
+
+  src = fetchurl {
+    url = "https:/www.worldpainter.net/files/${pname}_${version}.tar.gz";
+    hash = "sha256-FhrfI2/mBeIZAg26Q8x8T2YU2ANwAkoC3h9Tgw6ggoY=";
+  };
+
+  buildInputs = [
+    jre
+    gnused
+  ];
+
+  outputs = [ "out" ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,lib,.install4j/user}
+
+    install -Dm644 bin/*        "$out/bin/"
+    install -Dm644 lib/*        "$out/lib/"
+    install -Dm644 *.vmoptions  "$out/"
+    install -Dm755 worldpainter "$out/"
+    install -Dm755 wpscript     "$out/"
+    find .install4j/ -maxdepth 1 -type f -exec install -Dm644 {} "$out/.install4j/" \;
+    find .install4j/user/ -maxdepth 1 -type f -exec install -Dm644 {} "$out/.install4j/user/" \;
+
+    mkdir -p $out/share/{pixmaps,applications}
+    install -D -m 644 ${./worldpainter.png} $out/share/pixmaps/worldpainter.png
```

Comment:
This icon should be fetched from the repository or somewhere else. Do not put it in nixpkgs.